plugins/net_plugin/net_plugin.cpp

```diff
@@ -2015,15 +2034,16 @@ namespace eosio {
             return;
          }
          if( msg.network_version != network_version) {
-            c->backwards_compatibility = is_old_version (msg.network_version);
+            int16_t mnv = to_net_version(msg.network_version);
+            c->backwards_compatibility = mnv < net_version;
```

Comment:
Seems like you need to set net_version and check it as boolean backwards_compatibility will not be enough going forward.